### ISSUE TITLE
[no-unused-vars] - False positive - Unused arguments flagged in namespace declarations

### ISSUE THREAD

**Issue by scott-lc** [typescript-eslint/typescript-eslint#2835]
Prior to version 4.9 , this linted properly:

```TS
export namespace Alpha {
   export interface Beta {
     doSomething: (value: unknown) => void;   // All good
  }
}
```

As of version 4.9.0, a the same code is generates a `no-unused-vars` warning:

```TS
// Inside of a namespace, `no-unused-vars` is incorrectly flagged.
export namespace Alpha {
   export interface Beta {
     doSomething: (value: unknown) => void;   // <--- Incorrect v4.9 `no-unused-vars` warning
  }
}
```

However, the exact same interface, when exported outside of a namespace is fine:

```TS
// The exact same thing outside of a namespace is fine.
export interface Beta {
   doSomething: (value: unknown) => void;   // All good
}
```

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/no-unused-vars": "error"
  }
}
```

```TS
export namespace Alpha {
   export interface Beta {
     doSomething: (value: unknown) => void;   // <--- Incorrect v4.9 `no-unused-vars` warning
  }
}
```

**Expected Result**

No warning

**Actual Result**

```
'value' is defined but never used.  eslint@typescript-eslint/no-unused-vars
```

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `4.9.0` |
| `@typescript-eslint/parser`        | `4.9.0` |
| `TypeScript`                       | `4.1.2` |
| `ESLint`                           | `7.14.0` |
| `node`                             | `14.13.1` |

**Comment by bradzacher** [comment]
I'm unable to reproduce this against master.

![image](https://user-images.githubusercontent.com/7462525/100781901-856a3680-33c0-11eb-991c-bdbdd94a1373.png)

Perhaps there's a version mismatch somewhere?
Could you please create an isolated repro repo to help me nail this down?

**Comment by scott-lc** [comment]
Indeed.  Nothing `rm -rf node_modules && npm install` didn't fix.  Sorry for the noise.
